### ISSUE TITLE
Garbled UTF-8 subtitle (git-3c86bd2)

### ISSUE THREAD

**Issue by akanosora** [mpv-player/mpv#2268]
I tried to play a movie with UTF-8 subtitle however some of the words are garbled. Others are correctly displayed.

The garbled sentence in mpv
![screen shot 2015-08-30 at 12 29 36 pm](https://cloud.githubusercontent.com/assets/9272721/9568577/484ebe8e-4f13-11e5-9ae7-570e01ea8959.png)
The same sentence in VLC
![screen shot 2015-08-30 at 12 30 11 pm](https://cloud.githubusercontent.com/assets/9272721/9568578/4a404974-4f13-11e5-9943-c3f0244742cd.png)

**Comment by wm4** [comment]
Can you upload the subtitle file somewhere?

**Comment by akanosora** [comment]
You could download the subtitle file from here: http://sub.makedie.me/download/256593/the%20lunchbox.cht.zip

**Comment by wm4** [comment]
At the time position indicated by the VLC screenshot, I see this:
```
19
00:02:30,790 --> 00:02:32,690
只落一小撮就夠
```
Which looks like neither of those screenshots.

The file is definitely utf-8 and even has a BOM, so it shouldn't get mangled. Maybe there's a font problem?

**Comment by akanosora** [comment]
I believe the garbled one in mpv is this:

00:02:35,890 --> 00:02:39,090
這新食譜會對你有用

The font I used in Helvetica Sans GB W3 (pre-installed in OSX, copied to /.fonts/). It should support all Chinese-Japanese characters.

**Comment by wm4** [comment]
Whether this font is picked up at all pretty much depends how mpv is setup and to which libass version (the subtitle renderer) it is linked.

Can you get a playback log with ``-v``?

**Comment by akanosora** [comment]
It doesn't show me any error or warning:

```
[cplayer] Command line options: '-v' '/Volumes/Luna/Films/Brave/Brave.mkv'
[cplayer] mpv git-3c86bd2 (C) 2000-2015 mpv/MPlayer/mplayer2 projects
[cplayer]  built on Thu Aug 27 13:49:12 CEST 2015
[cplayer] ffmpeg library versions:
[cplayer]    libavutil       54.31.100
[cplayer]    libavcodec      56.58.100
[cplayer]    libavformat     56.40.101
[cplayer]    libswscale      3.1.101
[cplayer]    libavfilter     5.38.100
[cplayer]    libswresample   1.2.101
[cplayer] ffmpeg version: N-74678-g6701c92
[cplayer] 
[cplayer] Configuration: ./waf configure --enable-static-build --disable-x11 --disable-sdl1 --disable-sdl2
[cplayer] List of enabled features: apple-remote asm atomic-builtins atomics av-pix-fmt-mmal av-version-info avcodec-chroma-pos-api avframe-metadata avframe-skip-samples bsd-fstatfs build-date cocoa coreaudio cplayer debug-build dlopen dvdnav dvdread encoding fchmod gl gl-cocoa glob iconv jpeg lcms2 libass libass-osd libav libavfilter libbs2b libdl libm libswresample lua nanosleep optimize osx-thread-name posix posix-or-mingw posix-spawn pthreads resampler shm static-build subprocess termios tv uchardet vapoursynth-core vapoursynth-lazy vda-default-init2 vda-gl vda-hwaccel videotoolbox-gl videotoolbox-hwaccel videotoolbox-vda-gl zlib
[global] config path: '' -> '/Users/Keny/.mpv'
[global] config path: 'mpv.conf' -/-> '/Users/Keny/.mpv/mpv.conf'
[global] config path: 'config' -> '/Users/Keny/.mpv/config'
[global] config path: 'mpv.conf' -> '/Applications/mpv.app/Contents/Resources/mpv.conf'
[global] config path: 'config' -/-> '/Applications/mpv.app/Contents/Resources/config'
[global] config path: 'mpv.conf' -/-> '/usr/local/etc/mpv/mpv.conf'
[global] config path: 'config' -/-> '/usr/local/etc/mpv/config'
[cplayer] Reading config file /Applications/mpv.app/Contents/Resources/mpv.conf
[cplayer] Setting option 'profile' = 'pseudo-gui' (flags = 4)
[cplayer] Setting option 'terminal' = 'no' (flags = 4)
[cplayer] Setting option 'force-window' = 'yes' (flags = 4)
[cplayer] Setting option 'idle' = 'once' (flags = 4)
[cplayer] Setting option 'screenshot-directory' = '~~desktop/' (flags = 4)
[cplayer] Reading config file /Users/Keny/.mpv/config
[cplayer] Setting option 'vo' = 'opengl-hq:interpolation' (flags = 4)
[cplayer] Setting option 'hwdec' = 'auto' (flags = 4)
[cplayer] Setting option 'vf-clr' = '' (flags = 4)
[cplayer] Setting option 'autofit-larger' = '95%x85%' (flags = 4)
[cplayer] Setting option 'ao' = 'coreaudio' (flags = 4)
[cplayer] Setting option 'audio-channels' = 'downmix' (flags = 4)
[cplayer] Setting option 'alang' = 'jp,en' (flags = 4)
[cplayer] Setting option 'sub-codepage' = 'utf8:gb18030' (flags = 4)
[cplayer] Setting option 'sub-text-font' = 'Hiragino Sans GB W3' (flags = 4)
[cplayer] Setting option 'osd-font' = 'Hiragino Sans GB W3' (flags = 4)
[cplayer] Setting option 'sub-text-font-size' = '45' (flags = 4)
[cplayer] Setting option 'osd-font-size' = '45' (flags = 4)
[cplayer] Setting option 'sub-text-border-size' = '1.5' (flags = 4)
[cplayer] Setting option 'osd-border-size' = '1.5' (flags = 4)
[cplayer] Setting option 'sub-text-shadow-offset' = '1.5' (flags = 4)
[cplayer] Setting option 'sub-text-shadow-color' = '#60000000' (flags = 4)
[cplayer] Setting option 'sub-text-margin-x' = '45' (flags = 4)
[cplayer] Setting option 'sub-text-margin-y' = '45' (flags = 4)
[cplayer] Setting option 'slang' = 'jp,en' (flags = 4)
```

**Comment by akanosora** [comment]
I think I found out what caused the issue. It was caused by this line in my config file:

```
sub-codepage = utf8:gb18030
```

Removing the line solved the issue.

**Comment by lachs0r** [comment]
Please post the complete output.

**Comment by lachs0r** [comment]
Okay then.

**Comment by wm4** [comment]
That still shouldn't garble it, though.

**Comment by wm4** [comment]
Yep, was a regression, fixed in d60270e.
